cpp/inc/bond/core/transforms.h

```diff
@@ -377,7 +377,7 @@ class RequiredFieldValiadator
     {}
 
 private:
-    void MissingFieldException() const;
+    BOND_NORETURN void MissingFieldException() const;
```

Comment:
Consider marking both the declaration and the definition, in case some compilers need both or report a mismatch.